test/extended/util/url/url.go

```diff
@@ -37,23 +40,69 @@ func (ut *Tester) Close() {
 	ut.podName = ""
 }
 
+func (ut *Tester) Response(test *Test) *Response {
+	responses := ut.Responses(test)
+	if responses == nil {
+		return nil
+	}
+	if len(responses) == 0 {
+		return nil
+	}
+	return responses[0]
+}
+
 func (ut *Tester) Responses(tests ...*Test) []*Response {
-	script := testsToScript(tests)
 	if len(ut.podName) == 0 {
 		name, err := createExecPod(ut.client, ut.namespace, "execpod")
+		// exit even on error passthrough
 		o.Expect(err).NotTo(o.HaveOccurred())
 		ut.podName = name
 	}
+	// testToScript needs to run after creating the pod
+	// in case we need to rsync files for a post body
+	script := testsToScript(tests)
 	output, err := e2e.RunHostCmd(ut.namespace, ut.podName, script)
-	o.Expect(err).NotTo(o.HaveOccurred())
+	if !ut.errorPassThrough {
+		o.Expect(err).NotTo(o.HaveOccurred())
+	}
+	if err != nil {
+		return []*Response{
+			{
+				Error: fmt.Sprintf("%#v", err),
+			},
+		}
+
+	}
 	responses, err := parseResponses(output)
-	o.Expect(err).NotTo(o.HaveOccurred())
+	if !ut.errorPassThrough {
+		o.Expect(err).NotTo(o.HaveOccurred())
+	}
+	if err != nil {
+		return []*Response{
+			{
+				Error: fmt.Sprintf("%#v", err),
+				Body:  []byte(output),
+			},
+		}
+	}
 	if len(responses) != len(tests) {
-		o.Expect(fmt.Errorf("number of tests did not match number of responses: %d and %d", len(responses), len(tests))).NotTo(o.HaveOccurred())
+		errStr := fmt.Sprintf("number of tests did not match number of responses: %d and %d", len(responses), len(tests))
+		// exit even on error passthrough
+		o.Expect(fmt.Errorf(errStr)).NotTo(o.HaveOccurred())
```

Comment:
this change doesn't seem to serve any purpose, but it does add an extra sprintf call.

Reply:
ah a leftover from a prior iteration which did not make the cut ... will address